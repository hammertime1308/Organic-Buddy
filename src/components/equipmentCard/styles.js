import styled from 'styled-components/native';

import { COLOR, FONT } from '../../styles';

export const StyledCard = styled.TouchableOpacity`
  border-radius: 5px;
  margin: 10px 10px 0px 10px;
  border: 1px solid ${COLOR.BlackBorder};
  display: flex;
  flex-direction: column;

  backgroundColor: ${COLOR.White};
  shadowColor: ${COLOR.Black};
  shadowOpacity: 0.26;
  shadowOffset: { width: 0, height: 2};
  elevation: 3;
  overflow: hidden;
`;

export const Title = styled.Text`
  font-size: ${FONT.FontSizeHeader};
  font-weight: ${FONT.FontWeightBold};
  margin: 10px 15px 5px 15px;
`;

export const Description = styled.Text`
  font-size: ${FONT.FontSizeNormal};
  margin: 0px 15px 10px 15px;
`;

export const Price = styled.Text`
  font-size: ${FONT.FontSizeNormal};
  margin: 0px 0px 10px 15px;
  font-weight: ${FONT.FontWeightSlightBold};
`;

export const Type = styled.View`
  width: 10px;
  background-color: ${props => (props.type === 'Hire' ? 'yellow' : 'red')};
`;
