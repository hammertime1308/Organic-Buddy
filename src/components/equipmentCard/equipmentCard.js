import React from 'react';
import { Image, View } from 'react-native';

import { Type, StyledCard, Title, Description, Price } from './styles';

export const EquipmentCard = ({
  id = '',
  equipmentName = 'Title',
  equipmentDescription = 'Description of equipment',
  equipmentPrice = 0,
  equipmentImages = [
    'https://cdn.croptrust.org/wp/wp-content/uploads/2014/12/about-us.jpg',
  ],
  equipmentType = '',
  sellerId = '',
  onPress = () => {},
}) => {
  return (
    <StyledCard id={id} onPress={onPress} activeOpacity={0.75}>
      <View style={{ display: 'flex', flexDirection: 'row' }}>
        <Type type={equipmentType} />
        <View style={{ flex: 1 }}>
          <Image
            style={{
              width: '100%',
              height: 120,
              resizeMode: 'cover',
            }}
            source={{
              uri: `${equipmentImages[0]}`,
            }}
          />
          <Title>{equipmentName}</Title>
          <Description>{equipmentDescription}</Description>
          <View
            style={{ flexDirection: 'row', justifyContent: 'space-evenly' }}>
            <Price>Price: {equipmentPrice} Rs</Price>
            <Price>Ad Type: {equipmentType}</Price>
            <Price>Seller: {sellerId}</Price>
          </View>
        </View>
      </View>
    </StyledCard>
  );
};
