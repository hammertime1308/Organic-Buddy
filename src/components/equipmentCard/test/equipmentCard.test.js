import React from 'react';
import renderer from 'react-test-renderer';
import { EquipmentCard } from '../equipmentCard';

describe('Equipment card', () => {
  it('Should render card with image', () => {
    const tree = renderer
      .create(
        <EquipmentCard
          id="1"
          equipmentName="Sample title for testing"
          equipmentDescription="sample description for testing"
          equipmentPrice={1000}
          equipmentImages={[
            'https://cdn.pixabay.com/photo/2017/07/19/09/51/hallstatt-2518544_960_720.jpg',
          ]}
          equipmentType="Hire"
          sellerId="111"
          sellerContact="0123456789"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render card without image', () => {
    const tree = renderer
      .create(
        <EquipmentCard
          id="1"
          equipmentName="Sample title for testing"
          equipmentDescription="sample description for testing"
          equipmentPrice={1000}
          equipmentImages={[]}
          equipmentType="Hire"
          sellerId="111"
          sellerContact="0123456789"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render card for sell', () => {
    const tree = renderer
      .create(
        <EquipmentCard
          id="1"
          equipmentName="Sample title for testing"
          equipmentDescription="sample description for testing"
          equipmentPrice={1000}
          equipmentImages={[]}
          equipmentType="Sell"
          sellerId="111"
          sellerContact="0123456789"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
