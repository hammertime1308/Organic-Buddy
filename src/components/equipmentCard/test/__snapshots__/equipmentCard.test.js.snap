// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Equipment card Should render card for sell 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderColor": "rgba(0, 0, 0, 0.01)",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "elevation": 3,
      "flexDirection": "column",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "red",
            "width": 10,
          },
        ]
      }
      type="Sell"
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "700",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 10,
            },
          ]
        }
      >
        Sample title for testing
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "marginBottom": 10,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 0,
            },
          ]
        }
      >
        sample description for testing
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "fontWeight": "600",
                "marginBottom": 10,
                "marginLeft": 15,
                "marginRight": 0,
                "marginTop": 0,
              },
            ]
          }
        >
          Price: 
          1000
           Rs
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "fontWeight": "600",
                "marginBottom": 10,
                "marginLeft": 15,
                "marginRight": 0,
                "marginTop": 0,
              },
            ]
          }
        >
          Ad Type: 
          Sell
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Equipment card Should render card with image 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderColor": "rgba(0, 0, 0, 0.01)",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "elevation": 3,
      "flexDirection": "column",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "yellow",
            "width": 10,
          },
        ]
      }
      type="Hire"
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cdn.pixabay.com/photo/2017/07/19/09/51/hallstatt-2518544_960_720.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "resizeMode": "cover",
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "700",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 10,
            },
          ]
        }
      >
        Sample title for testing
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "marginBottom": 10,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 0,
            },
          ]
        }
      >
        sample description for testing
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "fontWeight": "600",
                "marginBottom": 10,
                "marginLeft": 15,
                "marginRight": 0,
                "marginTop": 0,
              },
            ]
          }
        >
          Price: 
          1000
           Rs
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "fontWeight": "600",
                "marginBottom": 10,
                "marginLeft": 15,
                "marginRight": 0,
                "marginTop": 0,
              },
            ]
          }
        >
          Ad Type: 
          Hire
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Equipment card Should render card without image 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderColor": "rgba(0, 0, 0, 0.01)",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "elevation": 3,
      "flexDirection": "column",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "yellow",
            "width": 10,
          },
        ]
      }
      type="Hire"
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "700",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 10,
            },
          ]
        }
      >
        Sample title for testing
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "marginBottom": 10,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 0,
            },
          ]
        }
      >
        sample description for testing
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "fontWeight": "600",
                "marginBottom": 10,
                "marginLeft": 15,
                "marginRight": 0,
                "marginTop": 0,
              },
            ]
          }
        >
          Price: 
          1000
           Rs
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 15,
                "fontWeight": "600",
                "marginBottom": 10,
                "marginLeft": 15,
                "marginRight": 0,
                "marginTop": 0,
              },
            ]
          }
        >
          Ad Type: 
          Hire
        </Text>
      </View>
    </View>
  </View>
</View>
`;
